iup/__init__.py

```diff
@@ -1111,24 +1055,13 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def validate(self):
-        estimate = self.select(pl.all().exclude(pl.Date))
-        SampleForecast.assert_column_name_all(estimate, "sample_id")
-
-    @staticmethod
-    def assert_column_name_all(estimate, column_name):
-        """
-        Verify that all columns have a pattern with a common name
+        # has at least 1 column of date and 1 column of estimate
+        self.assert_type_included(pl.Date)
+        self.assert_type_included(pl.Float64)
```

Comment:
I still think it would be better to check that the specific columns for the posterior samples are all type Float64, even if there are a lot of these columns. As long as the columns are named consistently (e.g. "Estimate_Sample_0001", "Estimate_Sample_0002", ... "Estimate_Sample_1000"), you could check that all columns beginning with "Estimate_Sample_" have type Float64.

Reply:
See the two new assert functions (8ead6e304c8f18304e32fbaa24b7f5b216ea247c), that check the name and type for each numerical column